Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp

```diff
@@ -205,14 +205,27 @@ static GstFlowReturn webkitMediaCommonEncryptionDecryptTransformInPlace(GstBaseT
     WebKitMediaCommonEncryptionDecrypt* self = WEBKIT_MEDIA_CENC_DECRYPT(base);
     WebKitMediaCommonEncryptionDecryptPrivate* priv = WEBKIT_MEDIA_CENC_DECRYPT_GET_PRIVATE(self);
 
+    GstProtectionMeta* protectionMeta = reinterpret_cast<GstProtectionMeta*>(gst_buffer_get_protection_meta(buffer));
+    bool isBufferEncrypted = protectionMeta;
+
     LockHolder locker(priv->m_mutex);
 
+    if (protectionMeta) {
+        const GValue* streamEncryptionEventsList = gst_structure_get_value(protectionMeta->info, "stream-encryption-events");
+        if (streamEncryptionEventsList && GST_VALUE_HOLDS_LIST(streamEncryptionEventsList)) {
+            unsigned streamEncryptionEventsListSize = gst_value_list_get_size(streamEncryptionEventsList);
+            for (unsigned i = 0; i < streamEncryptionEventsListSize; ++i)
+                priv->m_pendingProtectionEvents.append(GRefPtr<GstEvent>(static_cast<GstEvent*>(g_value_get_boxed(gst_value_list_get_value(streamEncryptionEventsList, i)))));
+            gst_structure_remove_field(protectionMeta->info, "stream-encryption-events");
+            isBufferEncrypted &= gst_structure_n_fields(protectionMeta->info) > 0;
```

Comment:
The &= is not needed, we know isBufferEncrypted will be true before this statement (although due to the aliasing with protectionMeta it's a bit more confusing than necessary imo), so it can be a straight assignment.

Reply:
I thought about this and I have my doubts too but I decided to do so because considering that there is no difference, we might add some code in the middle that could decide that it is not encrypted and then we would reset this if we are not careful enough, but it is a very long shot and agree with you.